code/modules/robotics/bot/secbot.dm

```diff
@@ -1152,16 +1156,29 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
 		src.name = "helmet/signaler/prox sensor/robot arm assembly"
 		src.overlays += image('icons/obj/bots/aibots.dmi', "hs_arm")
 		qdel(W)
-
-	else if ((istype(W, /obj/item/baton)) && (src.build_step >= 3))
-		src.build_step++
-		boutput(user, "You complete the Securitron! Beep boop.")
-		var/obj/machinery/bot/secbot/S = new /obj/machinery/bot/secbot(get_turf(src))
-		S.beacon_freq = src.beacon_freq
-		S.hat = src.hat
-		S.name = src.created_name
-		qdel(W)
-		qdel(src)
+		
+	else if (istype(W, /obj/item/rods) && src.build_step == 3)
+		if (W.amount < 1)
+			boutput(user, "You need a non-zero amount of rods. How did you even do that?")
+		else
+			src.build_step++
+			boutput(user, "You add a rod to [src]'s robot arm!")
+			src.name = "helmet/signaler/prox sensor/robot arm/rod assembly"
+			src.overlays += image('icons/obj/bots/aibots.dmi', "hs_rod")
+			W.amount -= 1
```

Comment:
Unfortunately, rods don't have a .use() proc. This means you'll need to manually check if the rod stack is empty and clear it out if it's gone:
```
				if (R.amount < 1)
					user.u_equip(R)
					qdel(R)
```

Reply:
My bad, forgot that rods don't always auto-delete themselves if their amount goes below 1. Changed!